File: app/src/main/java/com/dpis/module/config/ConfigSnapshotStore.java
package com.dpis.module.config;

import com.dpis.module.viewport.ViewportTargetSpec;

import java.util.Set;

public interface ConfigSnapshotStore {
    Set<String> getConfiguredPackages();

    boolean isTargetDpisEnabled(String packageName);

    ViewportTargetSpec getTargetViewportSpec(String packageName);

    String getTargetViewportApplyMode(String packageName);

    Integer getTargetFontScalePercent(String packageName);

    String getTargetFontApplyMode(String packageName);

    String getTargetTypefaceId(String packageName);

    String getPackageFontHookDomainsRaw(String packageName);

    boolean isSystemServerHooksEnabled();

    boolean isSystemServerSafeModeEnabled();

    boolean isGlobalLogEnabled();

    boolean hasSystemServerHooksEnabled();

    boolean hasSystemServerSafeModeEnabled();

    boolean hasGlobalLogEnabled();
}

File: app/src/main/java/com/dpis/module/config/ConfigSnapshotStore.kt
package com.dpis.module.config

import com.dpis.module.viewport.ViewportTargetSpec

interface ConfigSnapshotStore {
    fun getConfiguredPackages(): MutableSet<String?>
    fun isTargetDpisEnabled(packageName: String): Boolean
    fun getTargetViewportSpec(packageName: String): ViewportTargetSpec
    fun getTargetViewportApplyMode(packageName: String): String?
    fun getTargetFontScalePercent(packageName: String): Int?
    fun getTargetFontApplyMode(packageName: String): String?
    fun getTargetTypefaceId(packageName: String): String?
    fun getPackageFontHookDomainsRaw(packageName: String?): String?
    fun isSystemServerHooksEnabled(): Boolean
    fun isSystemServerSafeModeEnabled(): Boolean
    fun isGlobalLogEnabled(): Boolean
    fun hasSystemServerHooksEnabled(): Boolean
    fun hasSystemServerSafeModeEnabled(): Boolean
    fun hasGlobalLogEnabled(): Boolean
}
